terraform/load-balancer/variables.tf

```diff
@@ -0,0 +1,64 @@
+variable "basename" {
+  type        = string
+  description = "Basename of the module."
+  validation {
```

Comment:
Replace with the following validation:

validation {
  condition = can(regex("^[-\\.\\w]{0,77}$", var.basename)) && can(regex("[\\w]+$", var.basename))
  error_message = "The name must be between 0 and 77 characters, must end with a letter, number or underscore, and may contain only letters, numbers, underscores, periods, or hyphens."
  }

Reply:
Just implemented suggested changes. Resolving conversation. Re-open conversation if you have further suggestions or disagree with the changes proposed and implemented.